types/atto_fil.go

```diff
@@ -123,43 +102,39 @@ func NewAttoFILFromFILString(s string) (*AttoFIL, bool) {
 
 // NewAttoFILFromString allocates a new AttoFIL set to the value of s attofilecoin,
 // interpreted in the given base, and returns it and a boolean indicating success.
-func NewAttoFILFromString(s string, base int) (*AttoFIL, bool) {
-	af := NewZeroAttoFIL()
+func NewAttoFILFromString(s string, base int) (AttoFIL, bool) {
+	af := AttoFIL{}
 	_, ok := af.val.SetString(s, base)
 	return af, ok
 }
 
 // Add sets z to the sum x+y and returns z.
-func (z *AttoFIL) Add(y *AttoFIL) *AttoFIL {
-	ensureZeroAmounts(&z, &y)
-	newVal := big.NewInt(0)
-	newVal.Add(z.val, y.val)
-	newZ := AttoFIL{val: newVal}
-	return &newZ
+func (z AttoFIL) Add(y AttoFIL) AttoFIL {
+	newZ := AttoFIL{}
+	newZ.val.Add(&z.val, &y.val)
+	return newZ
 }
 
 // Sub sets z to the difference x-y and returns z.
-func (z *AttoFIL) Sub(y *AttoFIL) *AttoFIL {
-	ensureZeroAmounts(&z, &y)
-	newVal := big.NewInt(0)
-	newVal.Sub(z.val, y.val)
-	newZ := AttoFIL{val: newVal}
-	return &newZ
+func (z AttoFIL) Sub(y AttoFIL) AttoFIL {
+	newZ := AttoFIL{}
```

Comment:
so much clearer

Reply:
I don't think it would be "too clever" to use a named return value to reduce these to on-liners.